server/ui.go

```diff
@@ -0,0 +1,687 @@
+/*
+Copyright 2018 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"bytes"
+	"context"
+	"encoding/gob"
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"net/url"
+	"strings"
+
+	uuid "github.com/gofrs/uuid"
+	option "google.golang.org/api/option"
+	plus "google.golang.org/api/plus/v1"
+
+	"github.com/google/web-api-gateway/config"
+	"github.com/gorilla/mux"
+	"github.com/gorilla/securecookie"
+	"github.com/gorilla/sessions"
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/google"
+)
+
+const (
+	defaultSessionID     = "default"
+	profileSessionKey    = "profile"
+	oauthTokenSessionKey = "oauth_token"
+	stateSessionKey      = "state"
+	oauthFlowRedirectKey = "redirect"
+)
+
+var (
+	baseTmpl        = parseTemplate("")
+	listTmpl        = parseTemplate(*templatesFolder + "list.html")
+	editServiceTmpl = parseTemplate(*templatesFolder + "editService.html")
+	editAccountTmpl = parseTemplate(*templatesFolder + "editAccount.html")
+	keyTmpl         = parseTemplate(*templatesFolder + "key.html")
+	userTmpl        = parseTemplate(*templatesFolder + "user.html")
+	uploadTmpl      = parseTemplate(*templatesFolder + "upload.html")
+)
+
+var oauthConf *oauth2.Config = &oauth2.Config{
+	ClientID:     "523939206127-3pr1qbrn0g78l6r9nu10l733q9obgn0t.apps.googleusercontent.com",
+	ClientSecret: "zKY48Os4L8xKAuQoiBFqrLkW",
```

Comment:
what is this id/secret used for? do we need to further secure them in a key vault or something?

Reply:
according to the security review, it seems fine, i will keep an eye on this